app/services/comment_service.rb

```diff
@@ -50,12 +50,15 @@ def self.destroy(comment:, actor:)
 
   def self.update(comment:, params:, actor:)
     comment.edited_at = Time.zone.now
+    new_mentions = User.where username: Comment.new(params).mentioned_usernames - comment.mentioned_usernames
+
     comment.body = params[:body]
     return false unless comment.valid?
     actor.ability.authorize! :create, comment
     ThreadSearchService.index! comment.discussion_id
     comment.save!
     Memos::CommentUpdated.publish!(comment)
+    new_mentions.each { |user| Events::UserMentioned.publish!(comment, user) }
```

Comment:
Also, this is lightweight and readable enough that I don't really mind duplicating it across a few services (VoteService, MotionService, DiscussionService)

But we could do

```
MentionService.publish_new_mentions(discussion, params[:description], discussion.description])
```
```
def publish_new_mentions(model, new_value, old_value = "")
  return unless mentions = extract_mentioned_screen_names(new_value) - extract_mentioned_screen_names(old_value)
  User.where(username: mentions).each { |user| Events::UserMentioned.publish!(model, user)
end
```